### ISSUE TITLE
Failing tests do not fail the test suite

### ISSUE THREAD

**Issue by skeuchel** [andrewthad/quickcheck-classes#72]
The following run of the test suite has failing tests but overall the test suite passes.

```
$ cabal new-test -w ghc-7.10.3 --enable-tests --allow-older=aeson --constraint='aeson<0.9' .
<<<SHORTENED>>>
ToJSON/FromJSON: Partial Isomorphism *** Failed! Falsifiable (after 1 test):
0
ToJSON/FromJSON: Encoding Equals Value *** Failed! Falsifiable (after 1 test):
0
<<<SHORTENED>>>
One or more tests failed
Test suite basic: PASS
Test suite logged to:
/home/steve/trustee/quickcheck-classes/dist-newstyle/build/x86_64-linux/ghc-7.10.3/quickcheck-classes-0.6.0.0/t/basic/test/quickcheck-classes-0.6.0.0-basic.log
1 of 1 test suites (1 of 1 test cases) passed.
$ echo $?
0
```

**Comment by andrewthad** [comment]
This happens because we are currently using anything like `tasty` or `test-framework` in the test suite. I did this to make it easier for users to learn how to use this library by looking at the test suite. I just need to set the exit code appropriately. I'll have a fix up soon.

**Comment by andrewthad** [comment]
I fixed this in 5443f6ba16a7d4b24c3eceb30be76c120b9797cc. I accidentally committed it to master instead of making it a PR, but this addresses the issue you are having.

**Comment by skeuchel** [comment]
Thanks. Closing as it's resolved.
